src/main/java/uk/ac/ebi/ampt2d/accession/common/generators/ModelHashAccession.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package uk.ac.ebi.ampt2d.accession.common.generators;
+
+public interface ModelHashAccession<MODEL, HASH, ACCESSION> {
```

Comment:
It is not anonymous anymore.

Reply:
new ModelHashAccession<X,Y,Z>() { ... ?

Reply:
yes

Reply:
I mean that is an anonymous class. if we make ModelHashAccession a class instead of an interface, it will be smaller. On the other hand the usages of it will be maybe more ugly, e.g:
```
 .map(entry -> new ModelHashAccession<>(entry.getValue(), entry.getKey(),
                        generateAccessionFunction.apply(entry.getValue())))
```
instead of
```
.map(entry -> ModelHashAccession.of(entry.getValue(), entry.getKey(),
                        generateAccessionFunction.apply(entry.getValue())))
```
Like you said, this is constructor vs builder... I don't like how the interface ModelHashAccession looks, but I'm not super against it, so if anybody elses bother this, leave the interface